lib/phlex/translation.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module Phlex
+	module Translation
+		def self.included(view)
+			view.extend(ClassMethods)
+		end
+
+		module ClassMethods
+			attr_writer :translation_path
+
+			def translation_path
+				@translation_path ||= begin
+					source_location = const_source_location(name).first # Get view absolute file path
+					relative_path = source_location.split(Phlex.configuration.phlex_path).last # Get relative path to the phlex path
+					relative_path.delete_prefix("/").delete_suffix(".rb").tr("/", ".")
```

Comment:
@RolandStuder posted this related comment on the commit:

> You are relying on file paths here, though people might have custom autoloading setups, and have components / views outside of the `/app/views` directory. I think the better way would be to use `class.name.underscore` and use that as the "path", this would be more consistent with what i18n generally does, and it would be more flexible.
> 
> What do you think?

Reply:
I agree, `class.name.underscore` would be better. We could update the configuration to take a namespace to be removed from the beginning of the `class.name`.